code/modules/food_and_drinks/food/snacks_other.dm

```diff
@@ -789,6 +789,72 @@
 	trash = /obj/item/trash/can/food/peaches/maint
 	tastes = list("peaches" = 1, "tin" = 7)
 
+///How far a distance cats can usually get enticed by our delicious cat food
+#define CATFOOD_RANGE_NORMAL	6
+///With a rare chance, you'll summon the horde and get cats from all over
+#define CATFOOD_RANGE_EXTREME	20
+///The chance to summon cats from the above extremely high radius
+#define CATFOOD_EXTREME_CHANCE	1
+
+/obj/item/reagent_containers/food/snacks/canned/catfood
+	name = "cat food tin"
+	desc = "Purinya~ brand cat food, approved for both cat cats and people cats."
+	icon_state = "peachcan"
+	trash = /obj/item/trash/can/food/cat_food
+	tastes = list("organic slurry" = 5, "blended space carp" = 3)
+	///which cats are currently following us, lazylist
+	var/list/enticed_cats
+	///Cooldown for how long we're waiting to entice cats
+	COOLDOWN_DECLARE(catscan_cooldown)
+	///How long we wait to scan for the cats
+	var/catscan_delay = 5 SECONDS
+	///If we want to add a bonus to the normal entice range (get ALL the cats)
+	var/catscan_range_bonus = 0
+
+/obj/item/reagent_containers/food/snacks/canned/catfood/open_can(mob/user)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+	LAZYINITLIST(enticed_cats)
+
+/obj/item/reagent_containers/food/snacks/canned/catfood/Destroy()
+	. = ..()
+	for(var/i in enticed_cats)
+		var/mob/living/simple_animal/pet/cat/itercat = i
+		itercat.give_up_munchies()
+	LAZYCLEARLIST(enticed_cats)
+
+/obj/item/reagent_containers/food/snacks/canned/catfood/process()
+	if(!spillable || !COOLDOWN_FINISHED(src, catscan_cooldown))
+		return
+
+	var/catscan_range
+	var/list/scannables
+
+	//extreme scans don't care about visibility
+	if(prob(CATFOOD_EXTREME_CHANCE))
+		catscan_range = CATFOOD_RANGE_EXTREME
+		scannables = orange(catscan_range, get_turf(src))
+	else
+		catscan_range = CATFOOD_RANGE_NORMAL
+		scannables = ohearers(catscan_range, get_turf(src))
+
+	var/mob/living/simple_animal/pet/cat/itercat
+	for(itercat in scannables)
+		if(!itercat.stat && (!enticed_cats || !(itercat in enticed_cats)) && itercat.register_munchies(src))
+			LAZYADD(enticed_cats, itercat)
+
+	COOLDOWN_START(src, catscan_cooldown, catscan_delay)
+
+///Fake the cats munching on the food
+/obj/item/reagent_containers/food/snacks/canned/catfood/proc/cat_eat(mob/user)
+	reagents.remove_any(1)
```

Comment:
would it be possible to make these reagents react in some way with the cats that eat them? maybe use a manual call of both an ingestion reaction and an on_mob_life() effect (so that poisoning cat food will actually damage the cats who eat it)?

Reply:
i'm not snowflaking all that